app/src/main/java/com/android/streetworkapp/ui/progress/Progression.kt

```diff
@@ -0,0 +1,292 @@
+            }
+          }
+        }
+  }
+}
+
+@Composable
+fun CircularProgressBar(
+    progress: Float, // Value between 0 and 1
+    size: Dp,
+    strokeWidth: Dp = 12.dp,
+) {
+  Box(
+      contentAlignment = Alignment.Center,
+      modifier = Modifier.size(size).testTag("circularProgressBar")) {
+        Canvas(modifier = Modifier.size(size)) {
+          val sweepAngle = 360 * progress
+
+          // Background Arc
+          drawArc(
+              color = PROGRESSION_COLOR_GRAY,
+              startAngle = 0f,
+              sweepAngle = 360f,
+              useCenter = false,
+              style = Stroke(width = strokeWidth.toPx(), cap = StrokeCap.Round))
+
+          // Foreground Arc
+          drawArc(
+              color = PROGRESSION_COLOR_BLUE,
+              startAngle = -90f,
+              sweepAngle = sweepAngle,
+              useCenter = false,
+              style = Stroke(width = strokeWidth.toPx(), cap = StrokeCap.Round))
+        }
+
+        // Percentage Text
+        Text(
+            text = "${(progress * 100).toInt()}%",
+            fontSize = 26.sp,
+            fontWeight = FontWeight.Bold,
+            color = ColorPalette.PRIMARY_TEXT_COLOR)
+      }
+}
+
+@Composable
+fun MetricCard(label: String, value: String) {
+  Box(modifier = Modifier.padding(vertical = 4.dp)) {
+    Column(
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.Center,
+        modifier =
+            Modifier.border(width = 1.dp, color = Color.LightGray, shape = RoundedCornerShape(8.dp))
+                .size(120.dp, 50.dp)) {
+          Text(text = label, fontSize = 13.sp, color = ColorPalette.SECONDARY_TEXT_COLOR)
+          Text(
+              text = value,
+              fontSize = 15.sp,
+              fontWeight = FontWeight.Bold,
+              color = ColorPalette.PRIMARY_TEXT_COLOR)
+        }
+  }
+}
+
+@OptIn(ExperimentalLayoutApi::class)
+@Composable
+fun AchievementItem(achievement: Achievement) {
+  Row(
+      modifier = Modifier.fillMaxWidth().padding(vertical = 8.dp, horizontal = 16.dp),
+      verticalAlignment = Alignment.CenterVertically) {
+        Box(
+            modifier = Modifier.size(48.dp).background(Color.LightGray, shape = CircleShape),
+            contentAlignment = Alignment.Center) {
+              Image(
+                  painter = painterResource(id = achievement.icon),
+                  contentDescription = achievement.title,
+                  modifier = Modifier.size(64.dp).clip(CircleShape) // Clip the image to be circular
+                  )
+            }
+
+        Spacer(modifier = Modifier.width(16.dp))
+
+        Column(modifier = Modifier.weight(1f)) {
+          // Achievement Title
+          Text(
+              text = achievement.title,
+              fontSize = 16.sp,
+              fontWeight = FontWeight.Bold,
+              color = ColorPalette.PRIMARY_TEXT_COLOR)
+
+          // Tags Row
+          FlowRow(
```

Comment:
Suggestion: I think that `LazyRow` is more performant in this context

Reply:
Yes but this wouldn't allow for the tags to be on multiple lines. Keeping the FlowRow will keep the design more responsive.

Reply:
Given your argument conversation resolved